pkg/logql/log/labels.go

```diff
@@ -136,12 +234,154 @@ Outer:
 	return res
 }
 
-func (b *LabelsBuilder) WithoutLabels(names ...string) labels.Labels {
-	// naive implementation for now.
-	return b.Labels().WithoutLabels(names...)
+// LabelsResult returns the LabelsResult from the builder.
+// No grouping is applied and the cache is used when possible.
+func (b *LabelsBuilder) LabelsResult() LabelsResult {
```

Comment:
It would be great if we could somehow reuse `LabelsBuilder.Labels` function above because the only difference is call to `toResult` method.

Reply:
I can give a try to this.